lib/output.js

```diff
@@ -784,7 +784,9 @@ function OutputStream(options) {
         output.with_parens(function(){
             self.condition.print(output);
         });
-        output.semicolon();
+        if (output.option("beautify") && output.option("screw_ie8")) {
+            output.semicolon();
```

Comment:
Technically speaking this change gets rid of all trailing semi-colons after `do{...}while(...)`, some of which won't break IE8. But detecting those cases would make for more complicated code if I'm correct (still looking through the different code paths which make `{...}` blocks in `lib/output.js`)